src/components/Menu/index.js

```diff
@@ -2,57 +2,94 @@
 import React, { PureComponent } from 'react'
 import PropTypes from 'prop-types'
 
-import { isNull, map, isFunction, isUndefined } from 'lodash'
+import { isNull, map, isFunction, isUndefined, upperFirst } from 'lodash'
 import chain from 'helpers/generators/chain'
-import Item from './styles/Item'
-import ItemLink from './styles/ItemLink'
-import ItemsList from './styles/ItemsList'
-import StyledMenu from './styles/StyledMenu'
+
+import Styles from './styles'
 
 /**
  * The menu is a list of Link elements.
  *
  */
 class Menu extends PureComponent {
-  renderElement = element => {
+  getStylesFromType() {
+    const { type } = this.props
+
+    return Styles[upperFirst(type)]
+  }
+
+  renderElement = (element, isSelected) => {
     const { renderElement } = this.props
 
+    const { StyledItem, StyledItemLink } = this.getStylesFromType()
+
     return chain(
       // We assume a valid result as soon as we got a non null element
       element => !isNull(element) && !isUndefined(element),
       // Most precise, we got a custom render function for THAT element
       () =>
-        element && element.render && isFunction(element.render) ? element.render(element) : null,
+        element && element.render && isFunction(element.render)
+          ? element.render(element, isSelected)
+          : null,
       // A little less precise, we got a custom render function for each item
-      () => (renderElement && isFunction(renderElement) ? renderElement(element) : null),
+      () =>
+        renderElement && isFunction(renderElement) ? renderElement(element, isSelected) : null,
       // Least precise, neither other function returned something, we default to some hand made rendering
       () => (
-        <Item key={element.name}>
-          <ItemLink href={element.url} target={element.target} onClick={element.onClick}>
+        <StyledItem key={element.id || element.name}>
+          <StyledItemLink
+            href={element.url}
+            target={element.target}
```

Comment:
Please add the `rel="noreferrer noopener"` security to avoid `window.opener.location` hack in case `target == "_blank"`

Reply:
Fixed in link in upcoming commit

```
return (
      <StyledLink
        className={className}
        href={href}
        onClick={onClick}
        target={target}
        rel={target === '_blank' ? 'noreferrer noopener' : ''}
        {...rest}
      >
        {children}
      </StyledLink>
    )
```